l10n_nl_xaf_auditfile_export/models/xaf_auditfile_export.py

```diff
@@ -55,39 +62,30 @@ def _auditfile_name_get(self):
     company_id = fields.Many2one('res.company', 'Company', required=True)
 
     @api.model
-    def default_get(self, fields):
-        defaults = super(XafAuditfileExport, self).default_get(fields)
-        company = self.env['res.company'].browse([
-            self.env['res.company']._company_default_get(
-                object=self._model._name)])
-        fiscalyear = self.env['account.fiscalyear'].browse([
-            self.env['account.fiscalyear'].find(exception=False)])
-        if fiscalyear and self.env['account.fiscalyear'].search(
-                [('date_start', '<', fiscalyear.date_start)],
-                limit=1):
-            fiscalyear = self.env['account.fiscalyear'].search(
-                [('date_start', '<', fiscalyear.date_start)], limit=1,
-                order='date_stop desc')
-        if 'company_id' in fields:
+    def default_get(self, fields_list):
+        defaults = super(XafAuditfileExport, self).default_get(fields_list)
+        company = self.env.user.company_id
+        fy_dates = company.compute_fiscalyear_dates(datetime.now())
+        date_from = fields.Date.to_string(fy_dates['date_from'])
+        date_to = fields.Date.to_string(fy_dates['date_to'])
+        defaults.setdefault('date_start', date_from)
+        defaults.setdefault('date_end', date_to)
+        if 'company_id' in fields_list:
             defaults.setdefault('company_id', company.id)
-        if 'name' in fields:
+        if 'name' in fields_list:
             defaults.setdefault(
                 'name', _('Auditfile %s %s') % (
                     company.name,
-                    fiscalyear.name if fiscalyear
-                    else datetime.now().strftime('%Y')))
-        if 'period_start' in fields and fiscalyear:
-            defaults.setdefault('period_start', fiscalyear.period_ids[0].id)
-        if 'period_end' in fields and fiscalyear:
-            defaults.setdefault('period_end', fiscalyear.period_ids[-1].id)
+                    datetime.now().strftime('%Y')))
+
         return defaults
 
     @api.one
-    @api.constrains('period_start', 'period_end')
-    def check_periods(self):
-        if self.period_start.date_start > self.period_end.date_start:
+    @api.constrains('date_start', 'date_end')
+    def check_dates(self):
+        if self.date_start > self.date_end:
             raise exceptions.ValidationError(
-                _('You need to choose consecutive periods!'))
+                _('Starting date must be anterior ending date!'))
```

Comment:
According to the if condition, starting date can be also equal to ending date.
Please, either correct the condition or the error message.